diff --git a/configure.ac b/configure.ac
index 69ed043..d811071 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,6 +1,6 @@
 dnl Process this with autoconf to create configure
 
-AC_PREREQ([2.72])
+AC_PREREQ([2.68])
 
 AC_INIT([libffi],[3.5.0],[http://github.com/libffi/libffi/issues])
 AC_CONFIG_HEADERS([fficonfig.h])
@@ -94,10 +94,6 @@ m4_warn([obsolete],
 [The preprocessor macro `STDC_HEADERS' is obsolete.
   Except in unusual embedded environments, you can safely include all
   ISO C90 headers unconditionally.])dnl
-# Autoupdate added the next two lines to ensure that your configure
-# script's behavior did not change.  They are probably safe to remove.
-AC_CHECK_INCLUDES_DEFAULT
-AC_PROG_EGREP
 
 AC_CHECK_FUNCS(memcpy)
 AC_CHECK_HEADERS(alloca.h)
